Xamarin.Forms.Platform.Android/ImageCache.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Threading.Tasks;
+
+namespace Xamarin.Forms.Platform.Android
+{
+	/// <summary>
+	/// I setup the access to all the cache elements to be async because
+	/// if I didn't then it was locking up the GC and freezing the entire app
+	/// </summary>
+	class ImageCache
+	{
+
+		const string _cachePlaceHolder = "PLEASEHOLD";
+
+		global::Android.Util.LruCache _lruCache;
```

Comment:
Just some questions here...

What happens if `LruCache` evicts a `Bitmap` and it's in use? Does `Bitmap.Dispose` get called, _should_ it get called?

Related, will `Bitmap.Dispose` get called on a `Bitmap` that is still in the `LruCache` for reuse?

Reply:
> What happens if LruCache evicts a Bitmap and it's in use? Does Bitmap.Dispose get called, should it get called?

I did some research on this topic and found varying implementations. In theory we never retain drawables/bitmaps so if it's evicted and an imageview is collected the bitmap (having no references) should also be collected yes? A few implementations I saw involved people using custom drawables and then adding there own refcounters to them

https://stackoverflow.com/a/22855962/953734

> Related, will Bitmap.Dispose get called on a Bitmap that is still in the LruCache for reuse?

It shouldn't. We don't really call dispose on any of the bitmaps that get created from ResourceManager. It could be slightly breaking though if a user calls dispose so we could check the handle on the stored object as well?